### PR TITLE
[FX-4806] Create Fade and Backdrop components

### PR DESCRIPTION
**Pull request by TomasSlama**
[FX-4806]

### Description

Describe the changes and motivations for the pull request.

### How to test

- [Temploy](https://picasso.toptal.net/fx-4806-backdrop)
- Play with [Fade](https://picasso.toptal.net/fx-4806-backdrop/?path=/story/components-fade--fade) and [Backdrop](https://picasso.toptal.net/fx-4806-backdrop/?path=/story/components-backdrop--backdrop) stories

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [x] Make sure that additions and changes on the design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [x] Annotate all `props` in component with documentation
- [x] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [x] Covered with tests ([visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md) included)

**Breaking change**

- codemod is created and showcased in the changeset
- test alpha package of Picasso in StaffPortal

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-4806]: https://toptal-core.atlassian.net/browse/FX-4806?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/picasso/tsconfig.json [hunk 1]
```diff
@@ -48,6 +48,7 @@
     { "path": "../base/Pagination" },
     { "path": "../base/Paper" },
     { "path": "../base/PasswordInput" },
+    { "path": "../base/Popper" },
```
Comment (TomasSlama) [inline packages/picasso/tsconfig.json:51]: this change happened automatically after I run `refresh:tsconfig-references`

**Comment by TomasSlama** [comment]
@toptal-anvil ping reviewers

File: packages/picasso/tsconfig.json (hunk 1 above)
Comment (augustobmoura) [reply packages/picasso/tsconfig.json:51]: Hmm, it looks like it was outdated, I will open a PR so we can run the script on lint-staged

**Comment by TomasSlama** [comment]
@toptal-anvil ping reviewers

File: .storybook/components/CodeExample/CodeExample.tsx [hunk 2]
```diff
@@ -64,6 +64,8 @@ const imports: Record<string, object> = {
   '@toptal/picasso-account-select': require('@toptal/picasso-account-select'),
   '@toptal/picasso-alert': require('@toptal/picasso-alert'),
   '@toptal/picasso-amount': require('@toptal/picasso-amount'),
+  '@toptal/picasso-backdrop': require('@toptal/picasso-backdrop'),
+  '@toptal/picasso-fade': require('@toptal/picasso-fade'),
```
Comment (augustobmoura) [inline .storybook/components/CodeExample/CodeExample.tsx:68]: It's not about this PR in particular, but this whole object looks so egregious. Wdyt we refactor it before the other tailwind components?

This object is only used on the `resolver` function on line 138. We could replace that function with:
```ts
const resolver = (path: string) => {
  switch (path) {
    case 'react':
      return React

    case 'styled-components':
      return styled

    default:
      return path.startsWith('@toptal/picasso') ? require(path) : undefined
  }
}
```
And then remove this monstrosity :joy:

Comment (TomasSlama) [reply .storybook/components/CodeExample/CodeExample.tsx:68]: sure, good idea

Comment (TomasSlama) [reply .storybook/components/CodeExample/CodeExample.tsx:68]: @augustobmoura do you have any other comment related to this PR, or do you approve?

File: .storybook/components/CodeExample/CodeExample.tsx [hunk 3]
```diff
@@ -38,102 +38,20 @@ interface Props extends BaseProps {
...
+  'react-final-form',
+  'debounce',
+  'date-fns',
+  '@toptal/picasso',
+]
+
+const resolver = (path: string) =>
+  allowlistModulesPrefixes.some(allowedModule => path.startsWith(allowedModule))
+    ? require(path)
+    : undefined
```
Comment (OleksandrNechai) [inline .storybook/components/CodeExample/CodeExample.tsx:54]: If this is refactoring (which it looks like this to me) how is it related to creating Fade and Backdrop?

File: bin/refresh-tsconfig-references.mjs [hunk 4]
```diff
@@ -13,6 +13,20 @@ const formatFile = async file => {
   await $`yarn -s prettier --write ${file}`
 }
 
+const fileExists = async file => {
```
Comment (OleksandrNechai) [inline bin/refresh-tsconfig-references.mjs:16]: Where is this function used?

Comment (TomasSlama) [reply bin/refresh-tsconfig-references.mjs:16]: @augustobmoura added these changes. It is used when `yarn refresh:tsconfig-references` is called. Which is needed to be called after we create new component so the references are set correctly in the typescript config

File: .storybook/components/CodeExample/CodeExample.tsx (hunk 2 above)
Comment (augustobmoura) [reply .storybook/components/CodeExample/CodeExample.tsx:68]: We can add this change on a different PR, as it broke storybook while I as trying it

**Review by augustobmoura — APPROVED** [review]
Nice work!

**Pull request merged**